qiita_db/metadata_template.py

```diff
@@ -144,20 +144,32 @@ def _prefix_sample_names_with_id(md_template, study):
     study : Study
         The study to which the metadata belongs to
     """
-    # Create a new pandas series in which all the values are the study_id and
-    # it is indexed as the metadata template
-    study_ids = pd.Series([str(study.id)] * len(md_template.index),
-                          index=md_template.index)
-    # Create a new column on the metadata template that includes the metadata
-    # template indexes prefixed with the study id
-    md_template['sample_name_with_id'] = study_ids + '.' + md_template.index
-    # Assign the new previously created column as the new index
-    md_template.index = md_template.sample_name_with_id
-    # Delete the previously created column
-    del md_template['sample_name_with_id']
-    # The original metadata template had the index column unnamed - remove
-    # the name of the index
-    md_template.index.name = None
+    # Before we prefix the sample names with the study id, we should check if
+    # they have already prefixed with it
+    # Get all the prefixes of the index, defined as any string before a '.'
+    prefixes = set([idx.split('.')[0] for idx in md_template.index])
+    # If the samples have been already prefixed with the study id, the prefixes
+    # set will contain only one element and it will be the str representation
+    # of the study id
+    if len(prefixes) == 1 and prefixes.pop() == str(study.id):
+        warnings.warn("Sample names were already prefixed with the study id.")
+    else:
```

Comment:
This code is adding a sample/prep template to a study. If there are more
than one prefix, means that is a user-defined prefix, it is not the study
id prefix added by Qiita (which will be unique for all the samples) meaning
that we need to add the Qiita prefix, so the final sample id is
"<study_id>.<whatever.the.user.specifiied.as.sample.id>"

2015-02-11 13:19 GMT-08:00 Daniel McDonald <notifications@github.com>:

> In qiita_db/metadata_template.py
> <https://github.com/biocore/qiita/pull/843#discussion_r24534324>:
>
> > -    md_template.index = md_template.sample_name_with_id
> > -    # Delete the previously created column
> > -    del md_template['sample_name_with_id']
> > -    # The original metadata template had the index column unnamed - remove
> > -    # the name of the index
> > -    md_template.index.name = None
> > +    # Before we prefix the sample names with the study id, we should check if
> > +    # they have already prefixed with it
> > +    # Get all the prefixes of the index, defined as any string before a '.'
> > +    prefixes = set([idx.split('.')[0] for idx in md_template.index])
> > +    # If the samples have been already prefixed with the study id, the prefixes
> > +    # set will contain only one element and it will be the str representation
> > +    # of the study id
> > +    if len(prefixes) == 1 and prefixes.pop() == str(study.id):
> > +        warnings.warn("Sample names were already prefixed with the study id.")
> > +    else:
>
> Or that there are multiple prefixes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/843/files#r24534324>.
>



-- 
Jose Navas